### DISCUSSION TITLE
CLient lib like aiokafka

### DISCUSSION THREAD

**Discussion by sivaNbalusu5** [Aiven-Open/kio#95] [General]
Hi do you have any plans to create a library for clients like aiokafka?

**Comment by aiven-anton** [comment]
No there are really no such plans, and that's a much larger scope than what kio sets out to solve. [There's a summary of the project goals in the yet-to-be-published documentation.](https://github.com/Aiven-Open/kio/blob/main/docs/pages/goals.rst)

On the other hand, if the maintainers of aiokafka or kafka-python see the value in it, the opposite thing might happen in that they could start using kio to unlock new Kafka features, instead of further hand-crafting protocol entity representations.

(For posterity, I'm linking back to the relevant discussion on aiokafka: https://github.com/aio-libs/aiokafka/discussions/915#discussioncomment-6650096)
